src/camvidlog2/data.py

```
@@ -1,15 +1,49 @@
+# see https://stackoverflow.com/a/70917353/932342
+AnyEmbedding = Annotated[
+    Union[StringEmbedding, FrameEmbedding], Field(discriminator="source")
+]
+
+
+class EmbeddingGroup(BaseModel):
+    items: list[AnyEmbedding]
+
+
+def load_embedding_group_json(filename: str | Path) -> EmbeddingGroup:
+    with open(Path(filename), "r") as json_file:
```

Ruff (0.11.9)

Unnecessary mode argument

Remove mode argument

(UP015)